core/search/nested_filter.hpp

```diff
@@ -72,17 +74,28 @@ struct ByNestedOptions {
       return ((!lhs && !rhs) || (lhs && rhs && *lhs == *rhs));
     };
 
-    return match == rhs.match && merge_type == rhs.merge_type &&
-           equal(parent.get(), rhs.parent.get()) &&
-           equal(child.get(), rhs.child.get());
+    return match.index() == rhs.match.index() &&
+           std::visit(
+               [&]<typename T>(const T& v) noexcept -> bool {
+                 if constexpr (std::is_same_v<T, Match>) {
+                   return v == std::get<T>(rhs.match);
+                 }
+                 return true;
+               },
+               match) &&
+           merge_type == rhs.merge_type && equal(child.get(), rhs.child.get());
```

Comment:
you remove parent, is it ok?

Reply:
I have no idea how to compare 2 arbitrary `std::function<...>` reliably :man_shrugging: